elastic/kibana
# [Uptime] [Synthetics integration] browser monitors - Zip Url - add TLS Options and Proxy Url

**E2E testing not yet available**

Resolves #111482 

Adds TLS options for browser monitors when zip url is selected.

<img width="352" alt="Screen Shot 2021-10-13 at 7 37 50 PM" src="https://user-images.githubusercontent.com/11356435/137314328-bc355cd3-1990-4ade-9463-7f44fea5e435.png">
<img width="368" alt="Screen Shot 2021-10-13 at 7 37 40 PM" src="https://user-images.githubusercontent.com/11356435/137314329-2146e0f7-3b38-4ae8-b9a6-f4719e579f52.png">



_**Testing**_

**Setting up your e2e testing environment**
Overview: You will need to run a local version of kibana and package registry to test pending changes, but will use `elastic-package` to run a snapshot view of elasticsearch, fleet-server, and elastic-agent.  Since fleet server relies on kibana, we'll also use `elastic-package` to start a snapshot view of kibana exclusively for bootstrapping fleet-server, but will point our local version of kibana to a separate port. So we'll be running two versions of Kibana.
1. Stop all instances of elasticsearch or kibana
2. Checkout this PR. In `kibana.dev.yml` settings, add remove `kibana.index` and `xpack.task_manager.index` to remove legacy multitenancy features. This ensures that both our instances of Kibana can share saved objects related to fleet and agent. Also, add the following keys:
```
xpack.encryptedSavedObjects.encryptionKey: "12345678901234567890123456789012"
xpack.fleet.registryUrl: "http://localhost:8080"
server.port: 5602
```
4. Start your local kibana instance using `env BUILD_TS_REFS_DISABLE=true yarn kbn bootstrap && yarn start`
5. Update elastic package. It was recently updated to add support for the elastic-agent-complete image which contains support for running synthetics agent via heartbeat. Run `go get github.com/elastic/elastic-package`
6. Checkout https://github.com/elastic/integrations/pull/1778. Navigate to `packages/synthetics` in the repo directory. Run `elastic-package clean`, then `elastic-package build`.
7. Run `elastic-package stack up -d -v --services "elasticsearch,fleet-server,elastic-agent,package-registry" --version 8.0.0-SNAPSHOT` in the `packages/synthetics` directory. This will start up elasticsearch, fleet-server, elastic-agent, package registry AND kibana. This is because fleet-server requires kibana for bootstrapping.
8. Change the password to `kibana_system` user `curl -u elastic:changeme -X POST "http://localhost:9200/_security/user/kibana_system/_password?pretty" -H 'Content-Type: application/json' -d' { "password" : "changeme" } '`
9. Navigate to your local instance of kibana at **http://localhost:5602**

**Testing TLS**
1. Navigate to the synthetics integration at `fleet/integrations/synthetics-0.3.1/add-integration`
2. Add a policy name, then select the browser type from the dropdown
3. Fill out the form with the following options: Zip url: `https://github.com/elastic/synthetics-demo/archive/refs/heads/main.zip`, folder `todos/synthetics`, params `{
    "url": "https://elastic.github.io/synthetics-demo/"
}`
5. Toggle `Enable tls options for ZIP URL`. Remove all the version options from the ssl version combo box and add `TLSv1.0`. This will force the zip url to be fetched with an unsupported version.
<img width="356" alt="Screen Shot 2021-10-11 at 10 19 40 AM" src="https://user-images.githubusercontent.com/11356435/136873910-4fd88155-de15-456a-a65e-238a7fe3cd6a.png">
5. Save the integration and navigate to Uptime
6. Verify that the journey was not ran and data is not indexed into ES.
